### ISSUE TITLE
[Request] Batch batched graph

### ISSUE THREAD

**Issue by gybuay** [dmlc/dgl#1727]
## ❓ Questions and Help

Before proceeding, please note that we recommend
using our discussion forum (https://discuss.dgl.ai) for
general questions. As a result, this issue will
likely be CLOSED shortly.

I tried registering an account on your forum to post questions, but I never received a verification email, so I am here to seek answers.

One example data in my dataset consists of many small graphs. I use dgl.batch to batch them into a big graph. During training, assuming that my batch size is 32, how do I batch these batchedgraphs? I tried one_batch = dg.batch(big graph list), but it return all the small graphs if I do dg.unbatch(one_batch), where I desired big graphs to return.

**Comment by VoVAllen** [comment]
Sorry you can use `dgl.batch_hetero` for now and we will migrate this function with `dgl.batch` soon

**Comment by gybuay** [comment]
But my graphs are not heterogeneous.
A simple example is shown below.

`
g1 = dgl.DGLGraph()                           # B1 Graph 1
g1.add_nodes(2)
g1.ndata['h'] = torch.tensor([[1., 2.], [2., 3.]])
g2 = dgl.DGLGraph()                           # B1 Graph 2
g2.add_nodes(3)
g2.ndata['h'] = torch.tensor([[2., 3.], [3., 4.], [4., 5.]])

bg = dgl.batch([g1, g2], node_attrs='h')


g2_1 = dgl.DGLGraph()                           # B2 Graph 1
g2_1.add_nodes(2)
g2_1.ndata['h'] = torch.tensor([[1., 2.], [6., 9.]])
g2_2 = dgl.DGLGraph()                           # B2 Graph 2
g2_2.add_nodes(3)
g2_2.ndata['h'] = torch.tensor([[2., 3.], [3., 4.], [8., 9.]])

bg2 = dgl.batch([g2_1, g2_2], node_attrs='h')

# batch the two batchedgraphs
bgg = dgl.batch([bg,bg2], node_attrs='h')

# the below statement return 4 graphs, where I want bg1 and bg2 two objects
print(dgl.unbatch(bgg))
`


> Sorry you can use `dgl.batch_hetero` for now and we will migrate this function with `dgl.batch` soon

**Comment by VoVAllen** [comment]
Sorry my mistake. You if you batch two batchedgraph together, the original information will loss. You can create a bg3 batching those 4 graphs together also?

**Comment by jermainewang** [comment]
I think that's something we should support. We will work on a fix on that! The current workaround is like what  @VoVAllen said.

**Comment by gybuay** [comment]
> I think that's something we should support. We will work on a fix on that! The current workaround is like what @VoVAllen said.

Thanks for your solution. Looks like you are the author of this link https://zhuanlan.zhihu.com/p/55949794. I asked some questions about this article in comments yesterday and hope to get your answers.

**Comment by jermainewang** [comment]
Reopen for bookmarking this task.

**Comment by jermainewang** [comment]
Closed. This has been supported in 0.5.

**Comment by yazdanimehdi** [comment]
Hello,
It has the same functionality in the newer versions. Can you please help me to retain the first batching information in the second batching?
